src/attr.c

```diff
@@ -83,6 +84,55 @@ int git_attr_get(
 	return error;
 }
 
+int git_attr_get_by_tree(
```

Comment:
While binding this in rugged I noticed that the `repo` parameter here is not really necessary, we can always deduce the repo from the tree object.